src/components/UserAllowance.jsx

```diff
@@ -0,0 +1,168 @@
+import { Button, CircularProgress, Grid, TextField } from '@mui/material'
+import React, { useContext, useEffect, useState } from 'react'
+import { store } from 'store/store';
+import { useGetZoomAllowanceQuery } from '../hooks/useProfile'
+import { useQueryClient } from 'react-query'
+import { marketContractAddress, maxZOOMAllowance, QUERY_KEYS } from '../constants'
+import Slider from '@mui/material/Slider'
+import { styled as muiStyled } from '@mui/material/styles'
+
+const UserAllowanceWrapper = muiStyled('div')(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  padding: '16px',
+  alignItems: 'center',
+  maxWidth: '1024px',
+  margin: '0 auto',
+  marginLeft: '0px! important;',
+
+  '.slider-wrapper': {
+    display: 'flex',
+    alignItems: 'center',
+    margin: '12px 0px',
+
+    '.slider': {
+      paddingRight: '24px',
+    },
+  },
+
+  '.button-wrapper': {
+    marginTop: '16px',
+  },
+
+}))
+
+const UserAllowance = ({ initial }) => {
+  const {
+    state: { wallet, contracts },
+  } = useContext(store)
+  const { data: currentAllowance, isLoading } = useGetZoomAllowanceQuery(
+    wallet.address,
+    contracts.ZoomContract
+  )
+  const { zoomBalance } = wallet
+  const [zoomAllowance, setZoomAllowance] = useState(0)
+  const [isSettingAllowance, setIsSettingAllowance] = useState(false)
+
+  useEffect(() => {
+    if (initial !== undefined) {
+      setZoomAllowance(initial)
+    } else if (currentAllowance !== undefined) {
+      setZoomAllowance(currentAllowance)
+    }
+  }, [initial, currentAllowance])
+
+  const handleSliderChange = (event, newValue) => {
+    setZoomAllowance(newValue)
+  }
+
+  const handleInputChange = (event) => {
```

Comment:
If I clear the input, there's a console error. Prob left over from my previous implementation 😅 Can you handle this case please?

Reply:
Yep. I will do that!